### PR TITLE
enable CSEs for XMM operands

### PR DESCRIPTION
**Pull request by WalterBright**
It's past time to get common subexpressions working for XMM registers.

### CODE REVIEW

File: src/dmd/backend/cg87.d [hunk 1]
```diff
@@ -810,7 +810,11 @@ void fixresult87(ref CodeBuilder cdb,elem *e,regm_t retregs,regm_t *pretregs)
     /* if retregs needs to be transferred into the 8087 */
     if (*pretregs & mST0 && retregs & (mBP | ALLREGS))
     {
-        if (sz > DOUBLESIZE) elem_print(e);
+        if (sz > DOUBLESIZE)
+	{
+	    elem_print(e);
+	    printf("retregs = %s\n", regm_str(retregs));
+	}
```
Comment (thewilsonator) [inline src/dmd/backend/cg87.d]: Indentation?

**Review by thewilsonator — APPROVED** [review]

**Comment by WalterBright** [comment]
@wilzbach is there any way the scrolling window in semaphoreci can get an actual scroll tab larger than 2 pixels and also support paging up and down, and a larger window? It's nearly impossible to grab that tab and the only way to look around is to move the tab, and it's way too sensitive.

It makes it really, really hard to see what the problem is, as well as the actual problem may be a thousand lines back in the log. Why can't the failure be at the END of the log?

**Comment by thewilsonator** [comment]
I'd focus on the autotester:
```
Testing precise
timelimit -t 10 ./generated/linux/debug/64/precise --DRT-gcopt=gc:precise
    p = 0x7f502aa91000, size = 16, parent = (nil) 
    p = 0x7f502aa91010, size = 10, parent = (nil) 
    p = 0x7f502aa91000, size = 16, parent = (nil) 
    p = 0x7f502aa91010, size = 10, parent = (nil) 
    p = 0x7f502aa90000, size = 16, parent = (nil) 
    p = 0x7f502aa90000, size = 16, parent = (nil) 
    p = 0x7f5011988000, size = 272629760, parent = (nil) 
    p = 0x7f5025e88000, size = 68157440, parent = (nil) 
    p = 0x7f5011988000, size = 272629760, parent = (nil) 
    p = 0x7f5025e88000, size = 68157440, parent = (nil) 
    p = 0x7f502aa90000, size = 4113, parent = (nil) 
    p = 0x7f502aa90000, size = 4113, parent = (nil) 
posix.mak:300: recipe for target 'generated/linux/debug/64/unittest/libdruntime-ut.so' failed
make[1]: *** [generated/linux/debug/64/unittest/libdruntime-ut.so] Illegal instruction (core dumped)
make[1]: *** Waiting for unfinished jobs....
make[2]: Leaving directory '/media/disk1/braddr/sandbox/at-client/pull-3651640-Linux_32_64/druntime/test/gc'
make[1]: Leaving directory '/media/disk1/braddr/sandbox/at-client/pull-3651640-Linux_32_64/druntime'
posix.mak:270: recipe for target 'unittest-debug' failed
make: *** [unittest-debug] Error 2
```

**Comment by WalterBright** [comment]
I was looking at others hoping to find something more actionable than that.

**Comment by thewilsonator** [comment]
\*shrug\* you guess is a good as mine.

**Comment by rainers** [comment]
On semaphore, it's the same crash as for the autotester, i.e. building the druntime unittests AFAICT:
```
../dmd/generated/linux/release/32/dmd -conf= -Isrc -Iimport -w -de -dip1000 -preview=fieldwise -m64 -fPIC  -g -debug -version=Shared -fPIC -shared -unittest -ofgenerated/linux/debug/64/unittest/libdruntime-ut.so src/object.d ... src/etc/linux/memoryerror.d generated/linux/debug/64/errno_c.o generated/linux/debug/64/threadasm.o -L-ldl -debuglib= -defaultlib=
...
make[1]: *** [generated/linux/debug/64/unittest/libdruntime-ut.so] Illegal instruction (core dumped)
```
The messages immediately above are unrelated, they are just executed in parallel.

BTW: I agree the CI outputs are horrible, semaphore being the worst. One more reason to move to some custom build like d_do_test to hide the log of successful builds and tests.

**Comment by WalterBright** [comment]
@rainers thanks for the clue. I also found some clues in the buildkite logs, and corrected some of the issues.

**Comment by CyberShadow** [comment]
This pull request introduced a regression:
https://issues.dlang.org/show_bug.cgi?id=20349

**Pull request merged**